packages/astro/src/server/get-auth.ts

```diff
@@ -37,7 +42,13 @@ export const createGetAuth = ({ noAuthStatusMessage }: { noAuthStatusMessage: st
 
     const jwt = decodeJwt(authToken as string);
     // @ts-expect-error - TODO: Align types
-    return signedInAuthObject(options, jwt.raw.text, jwt.payload);
+    const authObject = signedInAuthObject(options, jwt.raw.text, jwt.payload);
+
+    if (treatPendingAsSignedOut && authObject.sessionStatus === 'pending') {
+      return signedOutAuthObject(options);
+    }
```

Comment:
This seems like something that _should_ be handled in `@clerk/backend`, but we can do it in a followup. Something like:

```
getAuthObjectFromJwt(jwt)
```

Reply:
Agreed! I'll raise a follow-up PR for it.